db/db_impl.h

```diff
@@ -652,13 +652,14 @@ class DBImpl : public DB {
   struct CompactionState;
 
   struct WriteContext {
-    autovector<SuperVersion*> superversions_to_free_;
+    SuperVersionContext superversion_context;
     autovector<MemTable*> memtables_to_free_;
 
+    explicit WriteContext(bool create_superversion = false)
```

Comment:
I guess you don't need the argument?

Reply:
We don't need it, but it may be useful for later optimizations.

Because the implementation of JobContext and WriteContext did not use the same code to deal with SVs, only JobContext was able to pre-allocate its SV, even if it may be useful for WriteContext as well.

I decided to disable it by default, but we might want to enable pre-allocation for some WriteContexts (maybe all of them). This is a bit outside of the scope of this PR, that's why I did not made any changes there, but if you think that it has its place here, I can investigate this.

We can't just blindly enable pre-allocation for all WriteContext, because I'm not sure that all the WriteContext are being used to install SVs and that would be a (small) waste of resource/time.

To sum up, a future improvement would be to check, for each WriteContext, if it is worth pre-allocating the SV or not.

Reply:
sounds good